ReleaseProcessScript/tools/Core/main_helper_functions.ps1

```diff
@@ -68,9 +68,29 @@ function Get-Hotfix-Current-Version ($StartReleasePhase)
   }
   else
   {
+    $HotfixBranchName = Get-Current-Branchname
     $MostRecentVersion = Get-Hotfix-Most-Recent-Version
 
-    $PossibleVersions = Get-Possible-Versions-Hotfix $MostRecentVersion $true
+    #Get current support branch from name and version of current branch
+    $CurrentBranchFullVersion = Parse-Version-From-BranchName $HotfixBranchName
+    $CurrentBranchMajorMinorVersion = Get-Major-Minor-From-Version $CurrentBranchFullVersion
+    $CurrentSupportBranchName = "support/v$($CurrentBranchMajorMinorVersion)"
+
+    # We want the following behaviour for the MeantForCurrentVersion flag of Get-Possible-Verions-Hotfix:
+    #
+    #   PreReleaseExists | NoReleaseExists |   MeantForCurrentVersion
+    #  ------------------|-----------------|-----------------------------
+    #   False            | False           | False
+    #   False            | True            | True
+    #   True             | False           | True
+    #   True             | True            | Impossible (fallback False)
+    #
+    # => this is equivalent to PreReleaseExists XOR NoReleaseExists
```

Comment:
I think I like it. And despite the double-negation for NoReleaseExists, the not-negated XOR is indeed a nice simplification.

Reply:
Regardless of the mathematical beauty, I do wonder if the Impossible case couldn't also do a fallback to True and thus make this into a simple OR setup. But that shouldn't hold up the merge.